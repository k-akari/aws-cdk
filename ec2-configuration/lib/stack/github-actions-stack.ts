import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { IamRole } from '../resource/iam-role';

export class GithubActionsStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    new IamRole().createGithubRole(this);
  }
}
