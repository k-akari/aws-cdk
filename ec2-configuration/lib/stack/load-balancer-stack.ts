import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc, CfnSecurityGroup } from 'aws-cdk-lib/aws-ec2';
import { CfnInstance } from 'aws-cdk-lib/aws-ec2';
import { Elb } from '../resource/elb';
import { Subnet } from '../resource/subnet';

export class LoadBalancerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, sg: CfnSecurityGroup, ec2Instance: CfnInstance, props?: StackProps) {
    super(scope, id, props);
   
    new Elb(this, vpc, subnet, sg, ec2Instance);
  }
}
