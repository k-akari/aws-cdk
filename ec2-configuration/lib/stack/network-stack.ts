import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { VPC } from '../resource/vpc';
import { Subnet } from '../resource/subnet';
import { InternetGateway } from '../resource/internet-gateway';
import { RouteTable } from '../resource/route-table';
import { NetworkAcl } from '../resource/network-acl';

export class NetworkStack extends Stack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new VPC(this);
    this.vpc = vpc.vpc;
    this.subnet = new Subnet(this, this.vpc);
    const internetGateway = new InternetGateway(this, this.vpc);
    new RouteTable(this, this.vpc, this.subnet, internetGateway.igw);
    new NetworkAcl(this, this.vpc, this.subnet);
  }
}
