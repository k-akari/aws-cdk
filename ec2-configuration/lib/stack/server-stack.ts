import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { Ec2 } from '../resource/ec2';
import { Subnet } from '../resource/subnet';
import { SecurityGroup } from '../resource/security-group';
import { IamRole } from '../resource/iam-role';
import { ElasticIp } from '../resource/elastic-ip';
import { Elb } from '../resource/elb';

export class ServerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, props?: StackProps) {
    super(scope, id, props);

    const sg = new SecurityGroup(this, vpc);

    const instanceProfile = new IamRole().createEc2InstanceProfile(this);
    const ec2Instance = new Ec2().createEc2Instance(this, subnet.public1a, instanceProfile, sg.ec2);
    new ElasticIp().associateElasticIp(this, ec2Instance, 'ec2-1a');

    new Elb(this, vpc, [subnet.public1a, subnet.public1c], sg.alb, ec2Instance);
  }
}
