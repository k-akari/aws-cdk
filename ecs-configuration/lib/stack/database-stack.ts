import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Subnet } from '../resource/subnet';
import { Rds } from '../resource/rds';
import { SecurityGroup } from '../resource/security-group';
import { SecretsManager } from '../resource/secrets-manager';

export class DatabaseStack extends Stack {
  constructor(scope: Construct, id: string, subnet: Subnet, sg: SecurityGroup, ssm: SecretsManager, props?: StackProps) {
    super(scope, id, props);
   
    new Rds(this, subnet, sg, ssm);
  }
}
