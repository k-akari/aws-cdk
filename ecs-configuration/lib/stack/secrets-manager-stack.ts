import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { SecretsManager } from '../resource/secrets-manager';

export class SecretsManagerStack extends Stack {
  public readonly ssm: SecretsManager

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    this.ssm = new SecretsManager(this);
  }
}
