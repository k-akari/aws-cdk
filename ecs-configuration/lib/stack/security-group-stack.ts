import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { SecurityGroup } from '../resource/security-group';

export class SecurityGroupStack extends Stack {
  public readonly sg: SecurityGroup;

  constructor(scope: Construct, id: string, vpc: Vpc, props?: StackProps) {
    super(scope, id, props);

    this.sg = new SecurityGroup(this, vpc);
  }
}
