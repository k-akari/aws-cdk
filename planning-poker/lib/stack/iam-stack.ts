import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { IamRole } from '../resource/iam-role';

export class IamStack extends Stack {
  public readonly iamRole: IamRole;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    this.iamRole = new IamRole(this);
  }
}
