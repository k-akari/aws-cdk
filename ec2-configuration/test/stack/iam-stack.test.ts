import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as Iam from '../../lib/stack/iam-stack';

test('IamStackSnapshot', () => {
  const serviceName = 'service';
  const envType = 'test';
  const app = new App({
    context: {
      'serviceName': serviceName,
      'envType': envType
    }
  });
  const stack = new Iam.IamStack(app, 'IamStack');
  const template = Template.fromStack(stack);

  expect(template).toMatchSnapshot();
});
