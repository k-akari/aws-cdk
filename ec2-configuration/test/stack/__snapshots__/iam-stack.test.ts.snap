// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IamStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "InstanceProfile": Object {
      "Properties": Object {
        "InstanceProfileName": "service-test-role-ec2",
        "Roles": Array [
          Object {
            "Ref": "RoleEc2",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "RoleEc2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "ec2.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore",
        ],
        "RoleName": "service-test-role-ec2",
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
